packages/components/src/VirtualizedList/HeaderCheckbox/HeaderCheckbox.component.js

```diff
@@ -0,0 +1,56 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import classnames from 'classnames';
+import theme from './HeaderCheckbox.scss';
+import getDefaultT from '../../translate';
+
+/**
+ * Header renderer that displays a "Select All" checkbox on header
+ */
+function HeaderCheckbox(props) {
+	const { columnData, t } = props;
+	const { id, onToggleAll, collection, isSelected } = columnData;
+	const isAllSelected = () =>
+		collection.length > 0 && collection.findIndex(item => !isSelected(item)) < 0;
+	return onToggleAll ? (
+		<form className={classnames('tc-list-checkbox', theme['tc-list-checkbox'])}>
+			<div className="checkbox" title={t('LIST_SELECT_ALL', { defaultValue: 'Select All' })}>
+				<label htmlFor={id && `${id}-header-check`}>
+					<input
+						id={id && `${id}-header-check`}
+						type="checkbox"
+						onChange={e => {
+							onToggleAll(e, collection);
```

Comment:
The collection comes from outside of the component, I think you can just call onToggleAll without passing the collection.
```
onChange={onToggleAll}
```